collector-sqs/src/main/java/zipkin/collector/sqs/SQSSpanProcessor.java

```diff
@@ -61,59 +61,74 @@
 
   @Override public void close() throws IOException {
     if (closed.getAndSet(true)) throw new IllegalStateException("SQS processor is already closed.");
+    synchronized (this) {
+      if (runner != null && !runner.isInterrupted()) runner.interrupt();
```

Comment:
note that this slightly differs from others that block until the last in-flight message completes.  For example, a lot of our newer code sets a flag to use (instead of while(true)). close sets that flag and waits for whatever in-flight request to finish. when the loop comes around and sees the flag false it gracefully ends the thread.

IOTW, I suspect that later someone may request to change this code to do something similar or otherwise don't interrupt as a first resort.